pkg/provision/cleanup.go

```diff
@@ -0,0 +1,230 @@
+package provision
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zbus"
+	"github.com/threefoldtech/zos/pkg"
+	"github.com/threefoldtech/zos/pkg/storage/filesystem"
+	"github.com/threefoldtech/zos/pkg/stubs"
+	"github.com/threefoldtech/zos/pkg/zdb"
+	"golang.org/x/net/context"
+)
+
+func execute(cmd string, args ...string) ([]byte, error) {
+	exe := exec.Command(cmd, args...)
+	output, err := exe.CombinedOutput()
+	if err != nil {
+		return nil, errors.Wrapf(err, "execute '%v %+v': %v", cmd, args, string(output))
+	}
+
+	return output, nil
+}
+
+func lines(input []byte) []string {
+	input = bytes.TrimSpace(input)
+	if len(input) == 0 {
+		return nil
+	}
+
+	return strings.Split(string(input), "\n")
+}
+
+func must(input []byte, err error) []byte {
+	if err != nil {
+		panic(err)
+	}
+
+	return input
+}
+
+type container struct {
+	ID   string `json:"ID"`
+	Spec struct {
+		Root struct {
+			Path string `json:"path"`
+		} `json:"root"`
+		Mounts []struct {
+			Source string `json:"source"`
+		} `json:"mounts"`
+	} `json:"Spec"`
+}
+
+// we declare this method as a variable so we can
+// mock it in testing.
+var zdbConnection = func(id string) zdb.Client {
+	socket := fmt.Sprintf("unix://%s/zdb.sock", socketDir(id))
+	return zdb.New(socket)
+}
+
+func socketDir(containerID string) string {
+	return fmt.Sprintf("/var/run/zdb_%s", containerID)
+}
+
+// CleanupResources cleans up unused resources
+func CleanupResources() error {
```

Comment:
This is a very big function, can we split the functionality into smaller functions, this will also help understanding the logic if u use descriptive function names.